apps/dgst.c

```diff
@@ -413,6 +413,28 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
+const char * newline_escape_filename(const char *file, int * backslash)
+{
+    int i = 0, length = strlen(file);
+    char *file_cpy = app_malloc(length + 3, file);
```

Comment:
Blank line after variable declarations please.

Why `length + 3`?
If more than one newline is acceptable, I'd suggest counting them before allocating.

Reply:
Yes, doing an initial first pass to count the newlines will be my new approach.  This will be a lot more robust than a static set of memory allocated.